pkg/tracing/otlp/otlp.go

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) The Thanos Authors.
+// Licensed under the Apache License 2.0.
+
+package otlp
+
+import (
+	"context"
+
+	"github.com/thanos-io/thanos/pkg/tracing/migration"
+
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	"github.com/pkg/errors"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracehttp"
+	"go.opentelemetry.io/otel/sdk/resource"
+	tracesdk "go.opentelemetry.io/otel/sdk/trace"
+	_ "google.golang.org/grpc/encoding/gzip"
+	"gopkg.in/yaml.v2"
+)
+
+var (
+	TracingClientGRPC string = "grpc"
+	TracingClientHTTP string = "http"
+)
+
+// NewOTELTracer returns an OTLP exporter based tracer.
+func NewTracerProvider(ctx context.Context, logger log.Logger, conf []byte) (*tracesdk.TracerProvider, error) {
+	config := Config{}
+	if err := yaml.Unmarshal(conf, &config); err != nil {
+		return nil, err
+	}
+
+	var exporter *otlptrace.Exporter
+	var err error
+	switch config.ClientType {
```

Comment:
Always lower-case this or upper-case this to make it indifferent to the case of the letters?